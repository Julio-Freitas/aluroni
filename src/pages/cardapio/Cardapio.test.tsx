import { render, screen } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import Cardapio from '.';

describe('testing component Cardapio', () => {
  test('The page should be render', ()=> {
    const { container } = render(
      <BrowserRouter>
        <Cardapio />
      </BrowserRouter>
    );
    const title = screen.getAllByRole('heading')[0];
    expect(title.textContent).toEqual('Card√°pio');
    expect(container).toMatchSnapshot();
  });
});
