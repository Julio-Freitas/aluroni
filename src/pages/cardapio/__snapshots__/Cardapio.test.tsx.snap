// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing component Cardapio The page should be render 1`] = `
<div>
  <section
    class="cardapio"
  >
    <h3
      class="titulo"
    >
      Cardápio
    </h3>
    <div
      class="buscador"
    >
      <input
        type="search"
        value=""
      />
      <svg
        fill="none"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M18.319 14.4326C20.7628 11.2941 20.542 6.75347 17.6569 3.86829C14.5327 0.744098 9.46734 0.744098 6.34315 3.86829C3.21895 6.99249 3.21895 12.0578 6.34315 15.182C9.22833 18.0672 13.769 18.2879 16.9075 15.8442C16.921 15.8595 16.9351 15.8745 16.9497 15.8891L21.1924 20.1317C21.5829 20.5223 22.2161 20.5223 22.6066 20.1317C22.9971 19.7412 22.9971 19.1081 22.6066 18.7175L18.364 14.4749C18.3493 14.4603 18.3343 14.4462 18.319 14.4326ZM16.2426 5.28251C18.5858 7.62565 18.5858 11.4246 16.2426 13.7678C13.8995 16.1109 10.1005 16.1109 7.75736 13.7678C5.41421 11.4246 5.41421 7.62565 7.75736 5.28251C10.1005 2.93936 13.8995 2.93936 16.2426 5.28251Z"
          fill="currentColor"
          fill-rule="evenodd"
        />
      </svg>
    </div>
    <div
      class="cardapio__filtros"
    >
      <div
        class="filtros"
      >
        <button
          class="filtros__filtro"
        >
          <div
            class="filtros__filtro_ball"
          />
          Massas
        </button>
        <button
          class="filtros__filtro"
        >
          <div
            class="filtros__filtro_ball"
          />
          Carnes
        </button>
        <button
          class="filtros__filtro"
        >
          <div
            class="filtros__filtro_ball"
          />
          Combos
        </button>
        <button
          class="filtros__filtro"
        >
          <div
            class="filtros__filtro_ball"
          />
          Veganos
        </button>
      </div>
      <button
        class="ordenador"
      >
        <span
          data-testid="nomeOrdernador"
        >
          Ordernar por
        </span>
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 0h24v24H0V0z"
            fill="none"
          />
          <path
            d="M7.41 8.59L12 13.17l4.59-4.58L18 10l-6 6-6-6 1.41-1.41z"
          />
        </svg>
        <div
          class="ordenador__options"
          data-testid="container-options"
        >
          <div
            class="ordenador__option ordenador__option--ativo"
            data-testid=""
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M0 0h24v24H0V0z"
                fill="none"
              />
              <path
                d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zm2.46-7.12l1.41-1.41L12 12.59l2.12-2.12 1.41 1.41L13.41 14l2.12 2.12-1.41 1.41L12 15.41l-2.12 2.12-1.41-1.41L10.59 14l-2.13-2.12zM15.5 4l-1-1h-5l-1 1H5v2h14V4z"
              />
            </svg>
            Selecione uma opção
          </div>
          <div
            class="ordenador__option"
            data-testid="porcao"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M18.06 22.99h1.66c.84 0 1.53-.64 1.63-1.46L23 5.05h-5V1h-1.97v4.05h-4.97l.3 2.34c1.71.47 3.31 1.32 4.27 2.26 1.44 1.42 2.43 2.89 2.43 5.29v8.05zM1 21.99V21h15.03v.99c0 .55-.45 1-1.01 1H2.01c-.56 0-1.01-.45-1.01-1zm15.03-7c0-8-15.03-8-15.03 0h15.03zM1.02 17h15v2h-15z"
              />
            </svg>
            Porção
          </div>
          <div
            class="ordenador__option"
            data-testid="qtd_pessoas"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M16 11c1.66 0 2.99-1.34 2.99-3S17.66 5 16 5c-1.66 0-3 1.34-3 3s1.34 3 3 3zm-8 0c1.66 0 2.99-1.34 2.99-3S9.66 5 8 5C6.34 5 5 6.34 5 8s1.34 3 3 3zm0 2c-2.33 0-7 1.17-7 3.5V19h14v-2.5c0-2.33-4.67-3.5-7-3.5zm8 0c-.29 0-.62.02-.97.05 1.16.84 1.97 1.97 1.97 3.45V19h6v-2.5c0-2.33-4.67-3.5-7-3.5z"
              />
            </svg>
            Qtd. pessoas
          </div>
          <div
            class="ordenador__option"
            data-testid="preco"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 0h24v24H0V0z"
                fill="none"
              />
              <path
                d="M8 17h2v-1h1c.55 0 1-.45 1-1v-3c0-.55-.45-1-1-1H8v-1h4V8h-2V7H8v1H7c-.55 0-1 .45-1 1v3c0 .55.45 1 1 1h3v1H6v2h2v1zM20 4H4c-1.11 0-1.99.89-1.99 2L2 18c0 1.11.89 2 2 2h16c1.11 0 2-.89 2-2V6c0-1.11-.89-2-2-2zm0 14H4V6h16v12zm-6-8l2-2 2 2m0 4.25l-2 2-2-2"
              />
            </svg>
            Preço
          </div>
        </div>
      </button>
    </div>
    <div
      class="itens"
    >
      <div
        class="item"
        data-testid="item-1"
      >
        <div
          class="item__imagem"
        >
          <img
            alt="Macarrão mediterrâneo"
            data-testid="img"
            src="/assets/pratos/macarrao_mediterraneo.png"
          />
        </div>
        <div
          class="item__descricao"
        >
          <div
            class="item__titulo"
          >
            <h2
              data-testid="title"
            >
              Macarrão mediterrâneo
            </h2>
            <p
              data-testid="description"
            >
              Macarrão spaghetti com frutos-do-mar e especíarias, especialidade da cozinha. Atenção: Contém camarão e outros frutos-do-mar!
            </p>
          </div>
          <div
            class="item__tags"
          >
            <div
              class="item__tipo item__tipo__massas"
              data-testid="category"
            >
              Massas
            </div>
            <div
              class="item__porcao"
              data-testid="size"
            >
              400
               g
            </div>
            <div
              class="item__qtdpessoas"
              data-testid="serving"
            >
              Serve 
              2
            </div>
            <div
              class="item__valor"
              data-testid="price"
            >
              R$ 
              73
            </div>
          </div>
        </div>
      </div>
      <div
        class="item"
        data-testid="item-2"
      >
        <div
          class="item__imagem"
        >
          <img
            alt="Macarrão da casa"
            data-testid="img"
            src="/assets/pratos/macarrao_da_casa.png"
          />
        </div>
        <div
          class="item__descricao"
        >
          <div
            class="item__titulo"
          >
            <h2
              data-testid="title"
            >
              Macarrão da casa
            </h2>
            <p
              data-testid="description"
            >
              Macarrão com frutos-do-mar e especíarias, especialidade da cozinha. Atenção: Contém camarão e outros frutos-do-mar!
            </p>
          </div>
          <div
            class="item__tags"
          >
            <div
              class="item__tipo item__tipo__massas"
              data-testid="category"
            >
              Massas
            </div>
            <div
              class="item__porcao"
              data-testid="size"
            >
              400
               g
            </div>
            <div
              class="item__qtdpessoas"
              data-testid="serving"
            >
              Serve 
              2
            </div>
            <div
              class="item__valor"
              data-testid="price"
            >
              R$ 
              54
            </div>
          </div>
        </div>
      </div>
      <div
        class="item"
        data-testid="item-3"
      >
        <div
          class="item__imagem"
        >
          <img
            alt="Macarrão com chilli"
            data-testid="img"
            src="/assets/pratos/macarrao_com_chilli.png"
          />
        </div>
        <div
          class="item__descricao"
        >
          <div
            class="item__titulo"
          >
            <h2
              data-testid="title"
            >
              Macarrão com chilli
            </h2>
            <p
              data-testid="description"
            >
              Macarrão fetuccine com molho de chilli e especiárias.
            </p>
          </div>
          <div
            class="item__tags"
          >
            <div
              class="item__tipo item__tipo__massas"
              data-testid="category"
            >
              Massas
            </div>
            <div
              class="item__porcao"
              data-testid="size"
            >
              350
               g
            </div>
            <div
              class="item__qtdpessoas"
              data-testid="serving"
            >
              Serve 
              1
            </div>
            <div
              class="item__valor"
              data-testid="price"
            >
              R$ 
              58
            </div>
          </div>
        </div>
      </div>
      <div
        class="item"
        data-testid="item-4"
      >
        <div
          class="item__imagem"
        >
          <img
            alt="Salada da casa com salmão grelhado"
            data-testid="img"
            src="/assets/pratos/salada_da_casa.png"
          />
        </div>
        <div
          class="item__descricao"
        >
          <div
            class="item__titulo"
          >
            <h2
              data-testid="title"
            >
              Salada da casa com salmão grelhado
            </h2>
            <p
              data-testid="description"
            >
              Salada da casa (alface, rúcula, cenoura) com salmão.
            </p>
          </div>
          <div
            class="item__tags"
          >
            <div
              class="item__tipo item__tipo__carnes"
              data-testid="category"
            >
              Carnes
            </div>
            <div
              class="item__porcao"
              data-testid="size"
            >
              450
               g
            </div>
            <div
              class="item__qtdpessoas"
              data-testid="serving"
            >
              Serve 
              1
            </div>
            <div
              class="item__valor"
              data-testid="price"
            >
              R$ 
              42
            </div>
          </div>
        </div>
      </div>
      <div
        class="item"
        data-testid="item-5"
      >
        <div
          class="item__imagem"
        >
          <img
            alt="Bife com batatas"
            data-testid="img"
            src="/assets/pratos/bife_com_batatas.png"
          />
        </div>
        <div
          class="item__descricao"
        >
          <div
            class="item__titulo"
          >
            <h2
              data-testid="title"
            >
              Bife com batatas
            </h2>
            <p
              data-testid="description"
            >
              Bife de boi (150g) na chapa, com batatas fritas. Acompanha arroz e molhos.
            </p>
          </div>
          <div
            class="item__tags"
          >
            <div
              class="item__tipo item__tipo__carnes"
              data-testid="category"
            >
              Carnes
            </div>
            <div
              class="item__porcao"
              data-testid="size"
            >
              550
               g
            </div>
            <div
              class="item__qtdpessoas"
              data-testid="serving"
            >
              Serve 
              2
            </div>
            <div
              class="item__valor"
              data-testid="price"
            >
              R$ 
              86
            </div>
          </div>
        </div>
      </div>
      <div
        class="item"
        data-testid="item-6"
      >
        <div
          class="item__imagem"
        >
          <img
            alt="Picanha picada com legumes e massa"
            data-testid="img"
            src="/assets/pratos/picanha_picada.png"
          />
        </div>
        <div
          class="item__descricao"
        >
          <div
            class="item__titulo"
          >
            <h2
              data-testid="title"
            >
              Picanha picada com legumes e massa
            </h2>
            <p
              data-testid="description"
            >
              400g de picanha picada, com legumes grelhados e massa escolhida pelo cliente.
            </p>
          </div>
          <div
            class="item__tags"
          >
            <div
              class="item__tipo item__tipo__combos"
              data-testid="category"
            >
              Combos
            </div>
            <div
              class="item__porcao"
              data-testid="size"
            >
              850
               g
            </div>
            <div
              class="item__qtdpessoas"
              data-testid="serving"
            >
              Serve 
              3
            </div>
            <div
              class="item__valor"
              data-testid="price"
            >
              R$ 
              123
            </div>
          </div>
        </div>
      </div>
      <div
        class="item"
        data-testid="item-7"
      >
        <div
          class="item__imagem"
        >
          <img
            alt="Lasagna vegana (carne de soja)"
            data-testid="img"
            src="/assets/pratos/lasagna_vegana.png"
          />
        </div>
        <div
          class="item__descricao"
        >
          <div
            class="item__titulo"
          >
            <h2
              data-testid="title"
            >
              Lasagna vegana (carne de soja)
            </h2>
            <p
              data-testid="description"
            >
              Lasagna 100% vegana feita com carne de soja e massa especial.
            </p>
          </div>
          <div
            class="item__tags"
          >
            <div
              class="item__tipo item__tipo__veganos"
              data-testid="category"
            >
              Veganos
            </div>
            <div
              class="item__porcao"
              data-testid="size"
            >
              350
               g
            </div>
            <div
              class="item__qtdpessoas"
              data-testid="serving"
            >
              Serve 
              1
            </div>
            <div
              class="item__valor"
              data-testid="price"
            >
              R$ 
              45
            </div>
          </div>
        </div>
      </div>
      <div
        class="item"
        data-testid="item-8"
      >
        <div
          class="item__imagem"
        >
          <img
            alt="Salada especial"
            data-testid="img"
            src="/assets/pratos/salada_especial.png"
          />
        </div>
        <div
          class="item__descricao"
        >
          <div
            class="item__titulo"
          >
            <h2
              data-testid="title"
            >
              Salada especial
            </h2>
            <p
              data-testid="description"
            >
              Salada com abacate, manga, alface, grão de bico e muito mais.
            </p>
          </div>
          <div
            class="item__tags"
          >
            <div
              class="item__tipo item__tipo__veganos"
              data-testid="category"
            >
              Veganos
            </div>
            <div
              class="item__porcao"
              data-testid="size"
            >
              350
               g
            </div>
            <div
              class="item__qtdpessoas"
              data-testid="serving"
            >
              Serve 
              1
            </div>
            <div
              class="item__valor"
              data-testid="price"
            >
              R$ 
              37
            </div>
          </div>
        </div>
      </div>
      <div
        class="item"
        data-testid="item-9"
      >
        <div
          class="item__imagem"
        >
          <img
            alt="Almôndegas"
            data-testid="img"
            src="/assets/pratos/almondegas.png"
          />
        </div>
        <div
          class="item__descricao"
        >
          <div
            class="item__titulo"
          >
            <h2
              data-testid="title"
            >
              Almôndegas
            </h2>
            <p
              data-testid="description"
            >
              Porção de almôndegas com sour cream, cebola e tempêros. Acompanha arroz ou massa.
            </p>
          </div>
          <div
            class="item__tags"
          >
            <div
              class="item__tipo item__tipo__combos"
              data-testid="category"
            >
              Combos
            </div>
            <div
              class="item__porcao"
              data-testid="size"
            >
              740
               g
            </div>
            <div
              class="item__qtdpessoas"
              data-testid="serving"
            >
              Serve 
              3
            </div>
            <div
              class="item__valor"
              data-testid="price"
            >
              R$ 
              120
            </div>
          </div>
        </div>
      </div>
    </div>
  </section>
</div>
`;
